cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MdsKey.java

```diff
@@ -57,7 +58,8 @@ static String getMetadataKey(String type, byte[] rowKey) {
     } else if (type.equals(EntityIdKeyHelper.TYPE_MAP.get(ApplicationId.class))) {
       keySplitter.skipString();
       keySplitter.skipString();
-    } else if (type.equals(EntityIdKeyHelper.TYPE_MAP.get(DatasetId.class))) {
```

Comment:
can't this method be simplified in the same way as the one above in MdsHistoryKey?